File: MATLAB code/Run_with_python/gen_dataset.py

import sys

from U_Net.utils import gen_dataset as gen_dataset_py

if __name__ == "__main__" : 
    
    target_dir = sys.argv[1]
    music_dir = sys.argv[2]
    voice_dir = sys.argv[3]
    voice_amp_ratio = float(sys.argv[4])
    train_test_split = bool(sys.argv[5])

    print("Generating data...", end="")
    gen_dataset_py(
        target_dir, music_dir, voice_dir, 
        voice_amp_ratio, train_test_split
    )
    print("\t--> Done")

File: MATLAB code/Run_with_python/gen_dataset.m

% Args
% ------------------------------------------------------------------------------------- %
% 1. target_dir : path where generated dataset will be stored.                          %
% 2. music_dir  : path where music source were stored.                                  %
% 3. voice_dir  : path where voice source were stored.                                  %
%                                                                                       %
% 4. voice_amp_ratio    :   the ratio to fit the amplitude of voice with music source.  %
%                           if voice_amp_ratio=0, the voice will be large as the music. %
%                                                                                       %
% 5. train_test_split   :   the ratio to split test dataset.                            %
%                           if it's 0, test dataset won't be generated.                 %
% ------------------------------------------------------------------------------------- %

function gen_dataset(target_dir, music_dir, voice_dir, voice_amp_ratio, train_test_split)

    args = target_dir + " " + music_dir + " " + voice_dir + " "...
        + voice_amp_ratio + " " + train_test_split;

    args = "gen_dataset.py " + args;
    pyrunfile(args)
end
